adb_client.c

```diff
@@ -134,10 +134,11 @@ static void handle_open_frame(adb_client_t *client, apacket *p) {
 
     /* OPEN(local-id, 0, "destination") */
     if (p->msg.arg0 == 0 || p->msg.arg1 != 0) {
+        adb_hal_apacket_release(client, p);
         return;
     }
 
-    name[p->msg.data_length > 0 ? p->msg.data_length - 1 : 0] = 0;
+    name[p->msg.data_length] = 0;
```

Comment:
I think this `name` variable can be removed to simplify the logic. `adb_service_open()` would become:
```
adb_service_t *adb_service_open(adb_client_t *client, apacket *p)
```
and it's possible to get `name` again from `p->data`. `strcmp` can be replaced with `memcmp` to avoid enforcing `\x00`.

I can handle this change if you want.

Reply:
Sure, no problem.